trending_data/src/test/java/com/google/sps/CountryTest.java

```diff
@@ -0,0 +1,76 @@
+package com.google.sps.data;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Class to test that parsing of country file is correct. The file is in TSV format and each country
+ * has six corresponding fields. Also test that a Country object is constructed correctly, i.e. all
+ * the fields take appropriate values.
+ */
+@RunWith(JUnit4.class)
+public final class CountryTest {
+  private static final String EMPTY_STRING = "";
+  private static final String INIT = "\tCountry\tCO\tCOU\t0\t0\t0";
+  /** Use this to construct a Country() object */
+  private static final String ALBANIA = "Albania\tAL\tALB\t8\t41\t20";
+
+  private static final String[] ALBANIA_ARRAY = {"Albania", "AL", "ALB", "8", "41", "20"};
+
+  private static final String AL_NAME = "Albania";
+  private static final String AL_ALPHA2 = "AL";
+  private static final double AL_LAT = 41;
+  private static final double AL_LNG = 20;
+
+  /** Use this for floating point comparison. Math.abs( expected – actual ) <= delta */
+  private static final double delta = 0.001;
+
+  private static final int CORRECT_NUMBER_OF_FIELDS = 6;
+
+  private Country parser;
+
+  @Before
+  public void setUp() {
+    parser = new Country(INIT);
```

Comment:
This is a code smell here, the fact that you need to half-initialize the country here so that the tests can run (or more than half-initialize, initialize it with a default String). Instancing a Country is not very expensive, so in this case I'd instance it in the tests where you need it, and remove the setUp method.

(Ideally, the parse() method would be private, because users of the Country class don't need to know how the parsing works, and all these tests would only deal with calling the Country() constructor and checking the fields. If you still want to test the parse() method, it makes more sense for it to be a static method, as it's purely a function that takes an input and returns an output without needing or changing any state).

Reply:
I was thinking about making it static since it doesn't make much sense for each Country instance to have its own parse method